app/src/main/java/com/keylesspalace/tusky/adapter/StatusBaseViewHolder.java

```diff
@@ -399,17 +407,28 @@ private void setSpoilerText(String spoilerText, List<Emoji> emojis,
             if (getAdapterPosition() != RecyclerView.NO_POSITION) {
                 listener.onExpandedChange(isChecked, getAdapterPosition());
             }
-            content.setVisibility(isChecked ? View.VISIBLE : View.GONE);
+            this.setTextVisible(isChecked);
 
         });
-        content.setVisibility(expanded ? View.VISIBLE : View.GONE);
+        this.setTextVisible(expanded);
 
     }
 
     private void hideSpoilerText() {
         contentWarningDescription.setVisibility(View.GONE);
         contentWarningButton.setVisibility(View.GONE);
-        content.setVisibility(View.VISIBLE);
+        this.setTextVisible(true);
+    }
+
+    private void setTextVisible(boolean visible) {
+        if (visible) {
+            content.setText(this.contentText);
+            this.content.setVisibility(View.VISIBLE);
+        } else if (this.mentionsOnlyContentText == null) {
```

Comment:
I think this method is called prior to the `mentionsOnlyContentText` being created. So the first time this will always be null. You could try creating the `mentionsOnlyContentText` here instead of making it a member variable.